### ISSUE TITLE
Add product locations for each warehouse

### ISSUE THREAD

**Issue by BB2A-Anthony** [Dolibarr/dolibarr#21886]
### Feature Request

The goal is to be able to define a location of the products in the warehouse to facilitate its physical research

I propose to add it in the stock tab on the product sheets

**Comment by cfoellmann** [comment]
We have done this as an ExtraField.
But it does not have very much functionality other then lookup

**Comment by BB2A-Anthony** [comment]
> We have done this as an ExtraField. But it does not have very much functionality other then lookup

thereafter there could be stock management by location

**Comment by cfoellmann** [comment]
Here is a module https://github.com/OPEN-DSI/dolibarr_module_compartmentstracking that should do what you need but it has not been touched in a long time

**Comment by elsp1991** [comment]
In what table have you added the extrafield for the location?
I was thinking it should be ideally in the stock_movement table but is hard to access this info

**Comment by BB2A-Anthony** [comment]
> In what table have you added the extrafield for the location?
> I was thinking it should be ideally in the stock_movement table but is hard to access this info

I have not added the extrafield for moment

**Comment by josett225** [comment]
Hi @BB2A-Anthony and @cfoellmann and @philippe-opendsi 
Is this project either still alive or thru another initiative? 
Regards
Jose

**Comment by josett225** [comment]
Hi @eldy 
I have 3 customers that are looking for this type of solution. I want your advise/suggestions/directions here to see if we could pursue in extending the module started several years ago by @philippe-opendsi 
As I don't know what it is plan in the future after 18 for warehouse locations, I need to know if I invest in developing this one or working on another initiative you may know?

Thanks for your feedback. 

Regards

Jose

**Comment by cfoellmann** [comment]
I would be up for it.
Maybe opendsi (@philippe-opendsi ) is cool with touching up the module and submitting the functionality to dolibarr core as a core feature!?

**Comment by josett225** [comment]
FYI, I just talked to @aspangaro that works also in Open-Dsi. He will look into it to and comeback to us.
Based on OpenDSI answer, we will be able to move forward.
By the way, if people want to test it, I did a changed in the code (a wrong sql parameter) to make it work in v17.
Here is the file /custom/compartmentstracking/class/compartment.class.php line 196.
I changed e.label by e.lieu
`	function fetch( $rowid, $ref='' )
	{
		// MODIF JOSE $sql = "SELECT c.rowid, c.ref, c.datec, c.fk_entrepot, c.column, c.shelf, c.drawer, c.status, e.label AS entrepot_label";
		$sql = "SELECT c.rowid, c.ref, c.datec, c.fk_entrepot, c.column, c.shelf, c.drawer, c.status, e.lieu AS entrepot_label";`
		
This is an interesting implementation of what we expected.

So far, here are some of the features missing for now if I am not wrong:
- Allocated/move products that are in a warehouse to the compartment of this warehouse. This is a kind of Stock Transfer between Compartments.
- Shipment/reception with Warehouse with compartment management
- shipment/reception PDF files with warehouse/compartment information
